Wrong (default) `tsconfig.json` used even though custom one is provided in tsup config

I have my own tsconfig file path defined to be used for tsup
```ts
// tsup.config.ts
{
  entry: ['src/export-types/index.ts'],
  dts: { only: true },
  clean: true,
  tsconfig: 'tsconfig.build.json',
}
```
which has `incremental: false`
```ts
// tsconfig.build.json
{
  "compilerOptions": {
    ...
    "incremental": false,
    ...
}
```

Somehow tsup still pick up my (default) `tsconfig.json`, thats in the root of the project, which has `"incremental": true` and fails with an error:
```sh
CLI Building entry: src/export-types/index.ts
CLI Using tsconfig:tsconfig.build.json  <-- notice that tsup output in terminal states correct/defined tsconfig file
CLI tsup v6.5.0
CLI Using tsup config: tsup.config.ts
DTS Build start
error TS5074: Option '--incremental' can only be specified using tsconfig, emitting to single file or when option '--tsBuildInfoFile' is specified.
```

If I delete `tsconfig.json` in root folder (that of course must be present for VS code to pick it up, and for development config), tsup bundles with defined/correct `tsconfig.build.json` and no error.

--- a/package.json
+++ b/package.json
@@ -57,7 +57,7 @@
     "postcss-simple-vars": "6.0.3",
     "prettier": "2.5.1",
     "resolve": "1.20.0",
-    "rollup-plugin-dts": "5.0.0",
+    "rollup-plugin-dts": "5.1.0",
     "rollup-plugin-hashbang": "2.2.2",
     "strip-json-comments": "4.0.0",
     "svelte": "3.46.4",

--- a/src/rollup.ts
+++ b/src/rollup.ts
@@ -151,6 +151,7 @@ const getRollupConfig = async (
         jsonPlugin(),
         ignoreFiles,
         dtsPlugin.default({
+          tsconfig: options.tsconfig,
           compilerOptions: {
             ...compilerOptions,
             baseUrl: compilerOptions.baseUrl || '.',

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -1143,3 +1143,58 @@ test('override target in tsconfig.json', async () => {
     )
   ).rejects.toThrowError(`Top-level await is not available in the configured target environment ("es2018")`)
 })
+
+test(`custom tsconfig should pass to dts plugin`, async () => {
+  const { outFiles } = await run(
+    getTestName(),
+    {
+      'input.ts': `export const foo = { name: 'foo'}`, 
+      'tsconfig.json': `{
+        "compilerOptions": {
+          "baseUrl":".",
+          "target": "esnext",
+          "incremental": true
+        }
+      }`,
+      'tsconfig.build.json': `{
+        "compilerOptions": {
+          "baseUrl":".",
+          "target": "esnext"
+        }
+      }`,
+      'tsup.config.ts': `
+        export default {
+          entry: ['src/input.ts'],
+          format: 'esm',
+          tsconfig: './tsconfig.build.json',
+          dts: {
+            only: true
+          }
+        }
+      `,
+    })
+    expect(outFiles).toEqual(['input.d.ts'])
+})
+
+test(`should generate export {} when there are no exports in source file`, async () => {
+  const { outFiles, getFileContent } = await run(
+    getTestName(),
+    {
+      'input.ts': `const a = 'a'`, 
+      'tsconfig.json': `{
+        "compilerOptions": {
+          "baseUrl":".",
+          "target": "esnext",
+        }
+      }`,
+      'tsup.config.ts': `
+        export default {
+          entry: ['src/input.ts'],
+          format: 'esm',
+          dts: true
+        }
+      `,
+    })
+    expect(outFiles).toEqual(['input.d.ts', 'input.mjs'])
+    expect(await getFileContent('dist/input.d.ts')).toContain('export { }')
+})

